backend/src/main/java/com/beaplab/BeaplabEngine/controller/UserController.java

```diff
@@ -130,6 +130,31 @@ public ResponseEntity<UserDto> get(@PathVariable("id") String id) {
         return new ResponseEntity<UserDto>(userDto, HttpStatus.OK);
     }
 
+    /**
+     * handles a GET request for retrieving a user account name, first name, last name
+     * @param id
+     * @return ResponseEntity<String>
+     */
+    @PreAuthorize("hasRole('ROLE_ADMIN') or hasRole('ROLE_USER')")
+    @RequestMapping(value = "/rest/beapengine/user/{id}/accountName", method = RequestMethod.GET, produces = "application/json")
+    @ApiOperation(value = "Find User account name by ID", notes = "Finding a User account name, first name, and last name by input id and returning the result in an object of type UserDto", response = UserDto.class)
+    public ResponseEntity<JSONObject> getAccountNameById(@PathVariable("id") String id) {
+
+        logger.info("in UserController/user/{" + id + "}/accountName GET method");
+
+        UserDto userDto = userService.get(id);
```

Comment:
Now if we add the previous check if session is valid we know we are now valid to check that the user exists. We can do this by

grabbing session details and using that to get the user id

```suggestion
  SessionDetails sessionDetails = (SessionDetails) session.getAttribute("SESSION_DETAILS");

  UserDto userDto = userService.get(sessionDetails.getUserId());
```